contribute/design/1.x/apis/cstor_volume.md

```diff
@@ -0,0 +1,309 @@
+	Type   CStorVolumeConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=CStorVolumeConditionType"`
+	Status ConditionStatus          `json:"status" protobuf:"bytes,2,opt,name=status,casttype=ConditionStatus"`
+	// Last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty" protobuf:"bytes,3,opt,name=lastProbeTime"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+	// Unique, this should be a short, machine understandable string that gives the reason
+	// for condition's last transition. If it reports "ResizePending" that means the underlying
+	// cstorvolume is being resized.
+	// +optional
+	Reason string `json:"reason,omitempty" protobuf:"bytes,5,opt,name=reason"`
+	// Human-readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message,omitempty" protobuf:"bytes,6,opt,name=message"`
+}
+
+// CStorVolumeConditionType is a valid value of CStorVolumeCondition.Type
+type CStorVolumeConditionType string
+
+const (
+	// CStorVolumeResizing - a user trigger resize of pvc has been started
+	CStorVolumeResizing CStorVolumeConditionType = "Resizing"
+)
+
+// ConditionStatus states in which state condition is present
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionInProgress" means corresponding
+// condition is inprogress. "ConditionSuccess" means corresponding condition is success
+const (
+	// ConditionInProgress states resize of underlying volumes are in progress
+	ConditionInProgress ConditionStatus = "InProgress"
+	// ConditionSuccess states resizing underlying volumes are successfull
+	ConditionSuccess ConditionStatus = "Success"
+)
+```
+# Changes to existing types
+
+- Remvoe the `Status` field of the `CStorvolumeSpec`. This should not be too problematic but `Status` filed has been used to   set the initial `init` state of `CStorVolume` to be used in other SAAS application like openebs-director etc.
+
+## `CStorVolumeReplicas` APIs
+
+```go
+
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CStorVolumeReplica describes a cstor volume resource created as custom resource
+type CStorVolumeReplica struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CStorVolumeReplicaSpec   `json:"spec"`
+	Status            CStorVolumeReplicaStatus `json:"status"`
+	VersionDetails    VersionDetails           `json:"versionDetails"`
+}
+
+// CStorVolumeReplicaSpec is the spec for a CStorVolumeReplica resource
+type CStorVolumeReplicaSpec struct {
+	TargetIP string `json:"targetIP"`
+
+  Capacity string `json:"capacity"`
+
+  // ZvolWorkers represents number of threads that executes client IOs
+  ZvolWorkers *int32 `json:"zvolWorkers"`
```

Comment:
Can this be called as IOWorkerThreads?

Reply:
this is more of having a common name symmetric with cstor(zfs) side. if required chaging `spec` will be required some more work as part of upgrade task.

Reply:
Changed to `IOWorkers`